stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1548,63 +1544,97 @@ func _TFCOpSetAttrStringArray(_ op: CTFEOp,
 
 /// A TensorFlow device kind.
 public enum DeviceKind {
-  /// Default device.
-  case `default`
   /// Central processing units.
   case cpu
   /// Graphics processing units.
   case gpu
   // TODO: TPU?
 }
 
-internal extension _ExecutionContext {
-  var currentDeviceKind: DeviceKind {
-    return sync {
-      deviceScopes.last ?? .default
+@usableFromInline
+class _ThreadLocalState {
+  var deviceScopes: [(DeviceKind, UInt)?] = []
+
+  private static let key = { () -> pthread_key_t in
```

Comment:
- Adding explicit types to non-trivial static variable declarations reduces type inference complexity.
- Not specifying closure types is generally preferred.

```suggestion
  private static let key: pthread_key_t = {
```